lib/biker.rb

```diff
@@ -1,3 +1,26 @@
 class Biker
+  attr_reader :name, :max_distance, :rides, :acceptable_terrain
 
-end
\ No newline at end of file
+  def initialize(name, max_distance)
+    @name = name
+    @max_distance = max_distance
+    @rides = Hash.new {|ride, time| hash[key] = []}
+    @acceptable_terrain = []
+  end
+
+  def learn_terrain!(terrain)
+    @acceptable_terrain << terrain
+  end
+
+  def log_ride(ride, time)
+    if @acceptable_terrain.include?(ride.terrain) && ride.total_distance <= @max_disatance
```

Comment:
Just had a small typo here -- max_distance instead of disatance. So close to getting this working! Logic looks good